app/controllers/halls_controller.rb

```diff
@@ -24,8 +24,9 @@ def edit
 
   def create
     authorize Hall
-    @hall = Hall.new(hall_params)
-    if @hall.save
+    @hall = ::Halls::Create.new(hall_params).call
+
+    if @hall.valid?
       redirect_to hall_url(@hall), notice: 'Hall was successfully created.'
     else
       render :new, status: :unprocessable_entity
```

Comment:
Do you plan to pass error messages and inform users about the reasons of failures?